src/jit/inlinepolicy.cpp

```diff
@@ -458,30 +464,132 @@ double LegacyPolicy::determineMultiplier()
+        break;
+    case InlineCallsiteFrequency::BORING:
+        multiplier += 1.3;
+        JITDUMP("\nInline candidate callsite is boring.  Multiplier increased to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::WARM:
+        multiplier += 2.0;
+        JITDUMP("\nInline candidate callsite is warm.  Multiplier increased to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::LOOP:
+        multiplier += 3.0;
+        JITDUMP("\nInline candidate callsite is in a loop.  Multiplier increased to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::HOT:
+        multiplier += 3.0;
+        JITDUMP("\nInline candidate callsite is hot.  Multiplier increased to %g.", multiplier);
+        break;
+    default:
+        assert(!"Unexpected callsite frequency");
+        break;
+    }
 
-//------------------------------------------------------------------------
-// hasNativeCodeSizeEstimate: did this policy estimate native code size
-//
-// Notes: 
-//    Temporary scaffolding for refactoring work.
+#ifdef DEBUG
 
-bool LegacyPolicy::hasNativeSizeEstimate()
-{
-    return (inlStateMachine != nullptr);
+    int additionalMultiplier = JitConfig.JitInlineAdditionalMultiplier();
+
+    if (additionalMultiplier != 0)
+    {
+        multiplier += additionalMultiplier;
+        JITDUMP("\nmultiplier increased via JitInlineAdditonalMultiplier=%d to %g.", additionalMultiplier, multiplier);
+    }
+
+    if (inlCompiler->compInlineStress())
+    {
+        multiplier += 10;
+        JITDUMP("\nmultiplier increased via inline stress to %g.", multiplier);
+    }
+
+#endif // DEBUG
+
+    return multiplier;
 }
 
 //------------------------------------------------------------------------
 // determineNativeCodeSizeEstimate: return estimated native code size for
 // this inline candidate.
 //
-// Notes: 
+// Notes:
 //    Uses the results of a state machine model for discretionary
-//    candidates.  Should not be needed for forcded or always
+//    candidates.  Should not be needed for forced or always
 //    candidates.
 
 int LegacyPolicy::determineNativeSizeEstimate()
 {
+    // Should be a discretionary candidate.
+    assert(inlObservation == InlineObservation::CALLEE_IS_DISCRETIONARY_INLINE);
+    // Should have a valid state machine estimate.
+    assert(inlNativeSizeEstimate != NATIVE_SIZE_INVALID);
+
     return inlNativeSizeEstimate;
 }
+
+//------------------------------------------------------------------------
+// determineNativeCallsiteSizeEstimate: estimate native size for the
+// callsite.
+//
+// Arguments:
+//    methInfo -- method info for the callee
+//
+// Notes:
+//    Estimates the native size (in bytes, scaled up by 10x) for the
+//    call site. While the quiality of the estimate here is questionable
+//    (especially for x64) it is being left as is for legacy compatibility.
+
+int LegacyPolicy::determineCallsiteNativeSizeEstimate(CORINFO_METHOD_INFO* methInfo)
```

Comment:
Since I"ve systematically got this wrong I'd prefer to just fix everything in a followup change.